pdc/apps/auth/views.py

```diff
@@ -903,3 +911,204 @@ def destroy(self, request, *args, **kwargs):
+            if not resource:
+                return Response(data={'resource': ['This field is required.']},
+                                status=status.HTTP_400_BAD_REQUEST)
+            try:
+                if permission:
+                    resource_permission_list = models.ResourcePermission.objects.filter(resource__name=resource,
+                                                                                        permission__name=permission)
+                else:
+                    resource_permission_list = models.ResourcePermission.objects.filter(resource__name=resource)
+            except Http404:
+                return Response({'detail': 'resource_permission %s is not existed' % data},
+                                status=status.HTTP_404_NOT_FOUND)
+        else:
+            resource_permission_list = models.ResourcePermission.objects.all()
+
+        for obj in resource_permission_list:
+            member = self._get_users_and_groups(obj)
+            result.setdefault(obj.resource.name, OrderedDict()).setdefault(obj.permission.name, member)
+
+        return Response(OrderedDict(sorted(result.items())), status=status.HTTP_200_OK)
+
+    def retrieve(self, request, *args, **kwargs):
+        """
+        __Method__: GET
+
+        __URL__: $LINK:apiresourcepermissions-detail:resource}/{permission$
+
+        __Response__:
+
+            {
+                "resource": string,
+                "permission": string
+                "users": list,
+                "groups": list,
+            }
+        """
+        data = self.kwargs
+        resource = data.pop('resource__name')
+        permission = data.pop('permission__name')
+        try:
+            resource_permission = models.ResourcePermission.objects.get(resource__name=resource, permission__name=permission)
+        except Http404:
+            return Response({'detail': 'resource_permission %s is not existed' % data},
+                            status=status.HTTP_404_NOT_FOUND)
+        members = self._get_users_and_groups(resource_permission)
+        members['resource'] = resource
+        members['permission'] = permission
+        return Response(members, status=status.HTTP_200_OK)
+
+    def create(self, request, *args, **kwargs):
+        """
+        __Method__: POST
+
+        __URL__: $LINK:apiresourcepermissions-list$
+
+         __Data__:
+
+            {
+                "group": string,
+                "resource": string,
+                "permission": string
+            }
+
+        __Response__:
+
+            {
+                "id": int,
+                "group": string,
+                "resource": string,
+                "permission": string
+            }
+
+        __Info__:
+
+        Only the permission of Group can be created.
+        """
+        return super(APIResourcePermissionViewSet, self).create(request, *args, **kwargs)
+
+    def destroy(self, request, *args, **kwargs):
+        """
```

Comment:
OK. I asked @dmach a question in JIRA. If their functions and parameters are totally same, I don't quite understand the reason we do it again.